File: fake-store-quarkus/src/main/java/org/arielzarate/application/ProductUseCase.java
package org.arielzarate.application;

import lombok.AllArgsConstructor;
import org.arielzarate.domain.model.Product;
import org.arielzarate.domain.ports.in.ProductService;
import org.arielzarate.domain.ports.out.ProductProvider;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;

import java.util.List;

@AllArgsConstructor
@ApplicationScoped
public class ProductUseCase implements ProductService {

    private final ProductProvider productProvider;

    @Override
    public Product getProductById(Integer id) {
        return productProvider.findById(id).orElse(null);
    }

    @Override
    public List<Product> getAllProducts() {
        return productProvider.findAll();
    }

    @Override
    public Product createProduct(Product product) {
        return productProvider.save(product);
    }
}

File: api-fake-hexagonal/src/main/kotlin/com/arielZarate/fake_api_hexagonal/application/ProductUseCase.kt
package com.arielZarate.fake_api_hexagonal.application

import com.arielZarate.fake_api_hexagonal.domain.model.Product
import com.arielZarate.fake_api_hexagonal.domain.ports.`in`.ProductService
import com.arielZarate.fake_api_hexagonal.domain.ports.out.ProductProvider
import org.springframework.stereotype.Service


@Service
class ProductUseCase(
    // nota: podria crear un servicio en domain que maneje alguna logiva o validacion si hiciese falta
    private val productProvider: ProductProvider
) : ProductService {

    override fun getProductById(id: Int): Product? {
        return productProvider.findProductById(id)

    }

    override fun getAllProducts(): List<Product> {
        return productProvider.getAllProducts()
    }

    override fun createProduct(product: Product): Product {
        return productProvider.saveProduct(product);
    }

}